testdata/testoutput2

```diff
@@ -14716,7 +14716,19 @@ Failed: error 161 at offset 9: subpattern number is too big
 No match
 
 /(?(1)()\983040\2)/
-Failed: error 161 at offset 14: subpattern number is too big
+Failed: error 161 at offset 13: subpattern number is too big
+
+# larger than GROUP_MAX, smaller than INT_MAX
+/a\800000b/
```

Comment:
No change - test passes with no difference to expected

Reply:
the offset is incorrect though, should be 8 no?

Reply:
The changes required weren't as intrusive as I feared so they are included in the alternative PR to solve this issue.